File: Sort Colors - Leetcode 75/Sort Colors - Leetcode 75.js
var sortColors = function(nums) {
    let counts = [0, 0, 0];

    for (let color of nums) {
        counts[color]++;
    }

    let R = counts[0], W = counts[1], B = counts[2];

    nums.fill(0, 0, R);
    nums.fill(1, R, R + W);
    nums.fill(2, R + W, R + W + B);
};

File: Sort Colors - Leetcode 75/Sort Colors - Leetcode 75.cpp
#include <vector>

class Solution {
public:
    void sortColors(std::vector<int>& nums) {
        int counts[3] = {0, 0, 0};

        for (int color : nums) {
            counts[color]++;
        }

        int R = counts[0], W = counts[1], B = counts[2];

        for (int i = 0; i < R; i++) {
            nums[i] = 0;
        }
        for (int i = R; i < R + W; i++) {
            nums[i] = 1;
        }
        for (int i = R + W; i < nums.size(); i++) {
            nums[i] = 2;
        }
    }
};
